test/rekt/features/kafkasource/data_plane.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kafkasource
+
+import (
+	"context"
+
+	cetest "github.com/cloudevents/sdk-go/v2/test"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/reconciler-test/pkg/eventshub"
+	assert "knative.dev/reconciler-test/pkg/eventshub/assert"
+	"knative.dev/reconciler-test/pkg/feature"
+
+	"knative.dev/eventing-kafka/test/rekt/resources/kafkacat"
+	"knative.dev/eventing-kafka/test/rekt/resources/kafkasource"
+)
+
+// DataPlaneDelivery returns a feature testing if the event sent to Kafka is received by the sink
+// via KafkaSource installed before sending events to Kafka
+func DataPlaneDelivery(name string, ksopts []kafkasource.CfgFn, kcopts []kafkacat.CfgFn, matchers cetest.EventMatcher) *feature.Feature {
```

Comment:
part of this struggle is to think about what the invariant of the test is and what you want the test to be doing.

I would think about reframing the test more like: "Given this kafka source (it is made before the feature is invoked), and this event replayer instance (could be passed as a StepFn from `kafkacat.Install(name, kcopts...)` and then added to or called in a `Must`, did all the events get seen?

Reply:
So I might think to change the signature from 

```
func DataPlaneDelivery(name string, ksopts []kafkasource.CfgFn, kcopts []kafkacat.CfgFn, matchers cetest.EventMatcher) *feature.Feature {
```

to

```
func DataPlaneDelivery(name string, produceEvents feature.StepFn, matchers cetest.EventMatcher) *feature.Feature {
```

- No `ksopts []kafkasource.CfgFn`, because the source already exists.
- No `kcopts []kafkacat.CfgFn` because it is passed as an event production step function.

Reply:
@n3wscott Did you have time to give more thoughts on this?